crates/boot_srv/src/lib.rs

```diff
@@ -0,0 +1,150 @@
+/// - `GET /`
+///   - Response Body: `OkResponse | ErrResponse`
+///
+/// ##### 2.2. Publishing info to the boot server.
+///
+/// A `PUT` on `/boot/Base64Space/Base64Agent` with an `AgentInfoSigned`
+/// json object as the request body.
+///
+/// - The server MUST reject the request if the body is > 1024 bytes.
+/// - The server MUST reject the request if `createdAt` is not within
+///   3 minutes in either direction of the server time.
+/// - The server MUST reject the request if `expiresAt` is in the past.
+/// - The server MUST reject the request if `signature` is invalid vs
+///   the `agentInfo`.
+/// - If `isTombstone` is `true`, the server MUST delete any existing
+///   info being held.
+/// - If `isTombstone` is `false`, the server MAY begin storing the info.
+///   See section 3. on storage strategies below.
+///
+/// ##### 2.3. Listing data stored on the boot server.
+///
+/// A `GET` on `/boot/Base64Space`.
+///
+/// - The server MUST respond with a complete list of stored infos.
+/// - If there are no infos stored at this space, the server MUST return
+///   an empty list (`[]`).
+/// - The only reason a server MAY return an `ErrResponse` for this request
+///   is in the case of internal server error.
+///
+/// ##### 2.4. Health check.
+///
+/// A `GET` on `/`.
+///
+/// - The server, in general, SHOULD return `OkResponse` to this request.
+/// - The server MAY return `ErrResponse` for internal errors or some other
+///   inability to continue serving correctly.
+///
+/// #### 3. Storage Strategies
+///
+/// ##### 3.1. The Future
+///
+/// It is the intention to someday in the future to add a "trusted" strategy,
+/// that will be triggerd via a new api, perhaps `/registerTrust/Base64Space`.
+///
+/// Even when that API is implemented, however, the default strategy defined
+/// next will be used on all spaces that have not been registered with a
+/// different strategy. Therefore, we can proceed for now without consideration
+/// for any other future strategies.
+///
+/// ##### 3.2. The "Default" Storage Strategy
+///
+/// Assumptions:
+///
+/// - We don't want to store unbounded count infos in each space.
+/// - We like storing long-running reliable nodes.
+/// - We don't want to ONLY store long-running nodes to avoid eclipse scenarios.
+///
+/// The solution put forth here is to allocate half our storage space to
+/// long-running nodes, and the other half to whoever has most recently
+/// published an info. The strategy for accomplishing this is defined here:
+///
+/// Consider "the store" acting as a stack. It can be implemented in any manner,
+/// but the following strategy assumes new entries are added to the end,
+/// and when any entries are removed, the indexes of any items after them
+/// will be decrimented.
+///
+/// - The server SHOULD provide a configurable per-space max info count.
+///   For the duration of this document, that value will be called MAX_INFOS.
+///   It is recommended to default that value to `32`.
```

Comment:
why?

Reply:
See: https://github.com/holochain/kitsune2/pull/6#discussion_r1844999439

We need enough that you can reasonably bootstrap an app, but not so much that servers can't be maintained. We can, of course, adjust this recommendation post-testing.

Reply:
Maybe worth saying explicitly "to keep the maximum response size from a request around 32K"